Library/Homebrew/utils/pypi.rb

```diff
@@ -224,9 +227,18 @@ def self.update_python_resources!(formula, version: nil, package_name: nil, extr
         package_name = list_entry["package_name"]
         extra_packages = list_entry["extra_packages"]
         exclude_packages = list_entry["exclude_packages"]
+        dependencies = list_entry["dependencies"]
       end
     end
 
+    missing_dependencies = (dependencies || []).reject { |dependency| Formula[dependency].any_version_installed? }
```

Comment:
```suggestion
    missing_dependencies = Array(dependencies).reject { |dependency| Formula[dependency].any_version_installed? }
```

Any chance `Formula[dependency]` will fail and should be non-fatal?

Reply:
> Any chance `Formula[dependency]` will fail and should be non-fatal?

There is a chance of failure unless we audit the list of formulae. I should probably add that somewhere around https://github.com/Homebrew/brew/blob/master/Library/Homebrew/tap_auditor.rb#L54

However, I am not sure if there is a non-fatal scenario as a missing dependency usually means command will fail later down the line. Main reason to catch would be cleaner `odie` message rather than stacktrace.

Reply:
I went ahead and added specific `odie` message.

Also will follow up with audit in #16800